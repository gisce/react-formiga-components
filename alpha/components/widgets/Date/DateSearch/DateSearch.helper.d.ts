import dayjs, { Dayjs } from "dayjs";
export declare const getMomentValue: (value: [string, string]) => (dayjs.Dayjs | null)[];
export declare const convertMomentDateArrayToStringArray: (momentValues: Dayjs[]) => (string | null)[];
//# sourceMappingURL=DateSearch.helper.d.ts.map