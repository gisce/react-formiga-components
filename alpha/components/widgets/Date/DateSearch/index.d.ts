export * from './DateSearch';
//# sourceMappingURL=index.d.ts.map