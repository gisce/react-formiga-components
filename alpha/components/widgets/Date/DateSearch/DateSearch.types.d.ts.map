{"version":3,"file":"DateSearch.types.d.ts","sourceRoot":"","sources":["../../../../../src/components/widgets/Date/DateSearch/DateSearch.types.tsx"],"names":[],"mappings":"AAAA,OAAO,EAAE,aAAa,EAAE,MAAM,cAAc,CAAC;AAE7C,MAAM,MAAM,eAAe,GAAG,aAAa,CAAC,CAAC,MAAM,EAAE,MAAM,CAAC,CAAC,CAAC;AAE9D,eAAO,MAAM,iBAAiB,eAAe,CAAC"}