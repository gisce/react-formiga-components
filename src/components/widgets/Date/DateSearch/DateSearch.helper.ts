import moment, { Moment } from "moment";
import { defaultDateFormat } from "./DateSearch.types";

export const getMomentValue = (value: [string, string]) => {
  return (
    value && [
      value[0] ? moment(value[0]) : null,
      value[1] ? moment(value[1]) : null,
    ]
  );
};

export const convertMomentDateArrayToStringArray = (momentValues: Moment[]) => {
  const from = momentValues[0] ? momentValues[0].format(defaultDateFormat) : null;
  const to = momentValues[1] ? momentValues[1].format(defaultDateFormat) : null;
  return [from, to];
};
