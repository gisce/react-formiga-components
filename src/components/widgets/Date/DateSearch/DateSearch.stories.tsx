import React, { useEffect, useState } from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { DateSearch } from ".";

const category = "Date specific";

export default {
  title: "Components/Widgets/Date/DateSearch",
  component: DateSearch,
  argTypes: {
    onChange: {
      action: "onChange",
      table: {
        category,
      },
    },
    value: {
      table: {
        category,
      },
    },
  },
} as ComponentMeta<typeof DateSearch>;

const Template: ComponentStory<typeof DateSearch> = (args) => {
  const [value, setValue] = useState(args.value);

  useEffect(() => {
    setValue(args.value);
  }, [args.value]);

  return (
    <DateSearch
      {...args}
      value={value}
      onChange={(value: [string, string]) => {
        setValue(value);
        args.onChange?.(value);
      }}
    />
  );
};

export const Basic = Template.bind({});
Basic.args = {};

export const BasicWithValues = Template.bind({});
BasicWithValues.args = { value: ["2020-01-01", "2020-12-01"] };
