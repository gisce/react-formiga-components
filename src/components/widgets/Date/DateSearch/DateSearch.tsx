import React from "react";
import { DatePicker } from "antd";
import { DateSearchProps } from "./DateSearch.types";
import { Moment } from "moment";
import {
  convertMomentDateArrayToStringArray,
  getMomentValue,
} from "./DateSearch.helper";

const defaultDateFormat = "DD/MM/YYYY";

export const DateSearch = (props: DateSearchProps) => {
  const { value, onChange } = props;

  const momentValue = getMomentValue(value);

  return (
    <DatePicker.RangePicker
      data-datepicker-id="datepicker-input"
      allowEmpty={[true, true]}
      format={defaultDateFormat}
      value={momentValue as any}
      onChange={(momentValues: Moment[]) => {
        onChange(
          convertMomentDateArrayToStringArray(momentValues) as [string, string]
        );
      }}
    ></DatePicker.RangePicker>
  );
};
