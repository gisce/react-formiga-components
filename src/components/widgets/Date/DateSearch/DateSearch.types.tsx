import { ValueOnChange } from "components";

export type DateSearchProps = ValueOnChange<[string, string]>;

export const defaultDateFormat = "DD/MM/YYYY";
