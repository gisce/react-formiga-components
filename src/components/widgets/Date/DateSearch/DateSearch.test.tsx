import React from "react";
import { render } from "@testing-library/react";

import { composeStories } from "@storybook/testing-react";
import * as stories from "./DateSearch.stories";

describe("DateSearch", () => {
  const testCases = Object.values(composeStories(stories)).map((Story) => [
    Story.storyName!,
    Story,
  ]);

  test.each(testCases)("Renders %s story", async (_storyName, Story) => {
    const tree = render(<Story />);
    expect(tree.baseElement).toMatchSnapshot();
  });
});
