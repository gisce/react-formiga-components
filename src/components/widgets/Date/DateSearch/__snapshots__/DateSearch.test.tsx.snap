// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSearch Renders Basic story 1`] = `
<body>
  <div>
    <div
      class="ant-picker ant-picker-range"
      data-datepicker-id="datepicker-input"
    >
      <div
        class="ant-picker-input ant-picker-input-active"
      >
        <input
          autocomplete="off"
          placeholder="Start date"
          readonly=""
          size="12"
          value=""
        />
      </div>
      <div
        class="ant-picker-range-separator"
      >
        <span
          aria-label="to"
          class="ant-picker-separator"
        >
          <span
            aria-label="swap-right"
            class="anticon anticon-swap-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="swap-right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
              />
            </svg>
          </span>
        </span>
      </div>
      <div
        class="ant-picker-input"
      >
        <input
          autocomplete="off"
          placeholder="End date"
          readonly=""
          size="12"
          value=""
        />
      </div>
      <div
        class="ant-picker-active-bar"
        style="left: 0px; width: 0px; position: absolute;"
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</body>
`;

exports[`DateSearch Renders BasicWithValues story 1`] = `
<body>
  <div>
    <div
      class="ant-picker ant-picker-range"
      data-datepicker-id="datepicker-input"
    >
      <div
        class="ant-picker-input ant-picker-input-active"
      >
        <input
          autocomplete="off"
          placeholder="Start date"
          readonly=""
          size="12"
          value="01/01/2020"
        />
      </div>
      <div
        class="ant-picker-range-separator"
      >
        <span
          aria-label="to"
          class="ant-picker-separator"
        >
          <span
            aria-label="swap-right"
            class="anticon anticon-swap-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="swap-right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M873.1 596.2l-164-208A32 32 0 00684 376h-64.8c-6.7 0-10.4 7.7-6.3 13l144.3 183H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h695.9c26.8 0 41.7-30.8 25.2-51.8z"
              />
            </svg>
          </span>
        </span>
      </div>
      <div
        class="ant-picker-input"
      >
        <input
          autocomplete="off"
          placeholder="End date"
          readonly=""
          size="12"
          value="01/12/2020"
        />
      </div>
      <div
        class="ant-picker-active-bar"
        style="left: 0px; width: 0px; position: absolute;"
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
      <span
        class="ant-picker-clear"
      >
        <span
          aria-label="close-circle"
          class="anticon anticon-close-circle"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="close-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</body>
`;
